x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_setup_flyout.tsx

```diff
@@ -6,65 +6,76 @@
 import { SingleMetricViewerInitializer } from './single_metric_viewer_initializer';
 import type { MlApiServices } from '../../application/services/ml_api_service';
-import { getDefaultSingleMetricViewerPanelTitle } from './get_default_panel_title';
 
 export async function resolveEmbeddableSingleMetricViewerUserInput(
   coreStart: CoreStart,
-  data: DataPublicPluginStart,
+  services: { data: DataPublicPluginStart; share?: SharePluginStart },
   mlApiServices: MlApiServices,
   input?: Partial<SingleMetricViewerEmbeddableInput>
-): Promise<Partial<SingleMetricViewerEmbeddableUserInput>> {
-  const { overlays, ...startServices } = coreStart;
+): Promise<SingleMetricViewerEmbeddableUserInput> {
+  const {
+    http,
+    overlays,
+    application: { currentAppId$ },
+    ...startServices
+  } = coreStart;
+  const { data, share } = services;
   const timefilter = data.query.timefilter.timefilter;
 
   return new Promise(async (resolve, reject) => {
     try {
-      const { jobIds } = await resolveJobSelection(
-        coreStart,
-        data.dataViews,
-        input?.jobIds ? input.jobIds : undefined,
-        true
-      );
-      const title = input?.title ?? getDefaultSingleMetricViewerPanelTitle(jobIds[0]);
-      const { jobs } = await mlApiServices.getJobs({ jobId: jobIds.join(',') });
-
-      const modalSession = overlays.openModal(
+      const flyoutSession = overlays.openFlyout(
         toMountPoint(
           <KibanaContextProvider
             services={{
               mlServices: { mlApiServices },
+              data,
+              share,
               ...coreStart,
             }}
           >
             <SingleMetricViewerInitializer
+              data-test-subj="mlSingleMetricViewerEmbeddableInitializer"
+              mlApiServices={mlApiServices}
               bounds={timefilter.getBounds()!}
-              defaultTitle={title}
               initialInput={input}
-              job={jobs[0]}
               onCreate={(explicitInput) => {
-                modalSession.close();
-                resolve({
-                  jobIds,
-                  ...explicitInput,
-                });
+                flyoutSession.close();
+                resolve(explicitInput);
               }}
               onCancel={() => {
-                modalSession.close();
+                flyoutSession.close();
                 reject();
               }}
             />
           </KibanaContextProvider>,
           startServices
-        )
+        ),
+        {
+          type: 'push',
+          ownFocus: true,
+          size: 's',
+          onClose: () => {
+            flyoutSession.close();
+            reject();
+          },
+        }
       );
+      // Close the flyout when user navigates out of the current plugin
+      currentAppId$
+        .pipe(skip(1), takeUntil(from(flyoutSession.onClose)), distinctUntilChanged())
+        .subscribe(() => {
+          flyoutSession.close();
```

Comment:
as this logic has been copied from the anomaly swim lane initializer, shall we consolidate it in a single function, e.g. `initMlEmbeddalbeFlyout`?

Reply:
Agreed - this is going to be very similar boiler plate code for all the embeddables 👍

Reply:
It's definitely the case for _every_ embeddable, i.e. for anomaly charts as well.

Reply:
@alvarezmelissa87 important update! we need to track overlay a bit differently, you can look up in the recent change point detection PR, where I also update the swim lane edit and add flow https://github.com/elastic/kibana/blob/4419370b9cab014b851ef1ce5fccd0e5b4a85995/x-pack/plugins/ml/public/embeddables/anomaly_swimlane/anomaly_swimlane_setup_flyout.tsx#L64

it will automatically track if the user navigates away from the dashboard app and disables all controls on the active dashboard.

Reply:
Added in 7603fd4b17fff9107d2d8a2ceb04547a7bcd46ac 🙏